### ISSUE TITLE
Unable to add an array of relations to an ElementQuery

### ISSUE THREAD

**Issue by jerome2710** [craftcms/cms#6573]
### Description
This seems to be a continuous struggle which keeps coming back to me. I might be overlooking something or doing something wrong, but I find it hard to find the correct way to perform am combined relation in an ElementQuery.

ElementQuery's with relations are not uncommon:

```php
$entries = Entry::find()->section('someSection')->relatedTo($relationValue)->all();
```

Where `$relationValue` can be sorts of things, as mentioned as possible values here: https://craftcms.com/docs/3.x/entries-fields.html#querying-elements-with-entries-fields. A single ID or `Entry`-object is simple, but multiple seem to be a struggle.

For example:
```php
public function getEntries()
{
    $entryQuery = Entry::find()->section('someSection');

    $relatedTo = [];

    if ($categorySlug = \Craft::$app->getRequest()->get('category')) {
        if ($category = Entry::find()->section('categories')->slug($categorySlug)->one()) {
            $relatedTo[] = $category->id;
        }
    }

    if ($themeSlug = \Craft::$app->getRequest()->get('theme')) {
        if ($theme = Entry::find()->section('themes')->slug($themeSlug)->one()) {
            $relatedTo[] = $theme->id;
        }
    }

    if (!empty($relatedTo)) {
        if (count($relatedTo) > 1) {
            $entryQuery->relatedTo(array_merge(['and'], $relatedTo));
        } else {
            $entryQuery->relatedTo($relatedTo[0]);
        }
    }

    return $entryQuery->all();
}
```

Eventually, the `relatedTo` with multiple relations would be `['and', 1, 2]`. Executing this query now gives me an error with the following stack trace:
```
2020-08-06 16:36:41 [-][1][aaaaa][error][yii\base\ErrorException:2] yii\base\ErrorException: array_merge(): Argument #2 is not an array in /..../vendor/craftcms/cms/src/elements/db/ElementRelationParamParser.php:141
Stack trace:
#0 /..../vendor/craftcms/cms/src/web/ErrorHandler.php(75): yii\base\ErrorHandler->handleError(2, 'array_merge(): ...', '/vagrant/hardlo...', 141)
#1 [internal function]: craft\web\ErrorHandler->handleError(2, 'array_merge(): ...', '/vagrant/hardlo...', 141, Array)
#2 /..../vendor/craftcms/cms/src/elements/db/ElementRelationParamParser.php(141): array_merge(Array, 646)
#3 /..../vendor/craftcms/cms/src/elements/db/ElementRelationParamParser.php(105): craft\elements\db\ElementRelationParamParser->_subparse(646)
#4 /..../vendor/craftcms/cms/src/elements/db/ElementQuery.php(2221): craft\elements\db\ElementRelationParamParser->parse(Array)
#5 /..../vendor/craftcms/cms/src/elements/db/ElementQuery.php(1414): craft\elements\db\ElementQuery->_applyRelatedToParam()
```

I really don't get it. Especially when looking at `\craft\elements\db\ElementRelationParamParser::_subparse`, which merges `$relCriteria` on r.138 as an array and then performs `is_array()` on r.164. Of course, it would be an array by then.

I usually find some sort of solution, for example by splitting it into separate queries and then running `Entry::find()->ids(implode())`, but shouldn't this be easier?

### Additional info

- Craft version: 3.5.0
- PHP version: 7.2.24

**Comment by wsydney76** [comment]
This is a snippet from one of our projects that, if i get the question right, solves a similar problem:

```
$query = Entry::find()->section('screening')->site('*')->unique();
...
return $query->relatedTo([
    'and',
    ['targetElement' => $filmIds, 'field' => 'film'],
    ['targetElement' => $locationIds, 'field' => 'location']

]);
```

**Comment by brandonkelly** [comment]
Thanks for reporting that! This was indeed a bug with the relation param parsing logic. Just fixed it for the next release.

**Comment by brandonkelly** [comment]
Just released Craft 3.5.2 with a fix for `['and', X, Y, Z]` syntax.
